### ISSUE TITLE
Retained message is not sent for Subscribe to existing topic

### ISSUE THREAD

**Issue by jhor511** [emqtt/emqttd#1314]
#### Environment

EMQ Broker v2.1.2

#### Description

This relates to issue 352 (https://github.com/emqtt/emqttd/issues/352). It appears this issue is present again in v2.1.2.

If an MQTT Client Subscribes to a Topic (for which a subscription already exists in the Broker), then the Broker *does not* send the last Retained message for that topic. 

According to MQTT-3.8.4-3, "If a Server receives a SUBSCRIBE Packet containing a Topic Filter that is identical to an existing Subscription’s Topic Filter then it MUST completely replace that existing Subscription with a new Subscription. The Topic Filter in the new Subscription will be identical to that in the previous Subscription, although its maximum QoS value could be different. **Any existing retained messages matching the Topic Filter MUST be re-sent**, but the flow of publications MUST NOT be interrupted"

The existing retained messages matching the Topic Filter *are not* being sent.

The workaround for me is to Unsubscribe from the topic before Subscribing. This effectively gets the last Retained message for that topic.

Output from Python program to recreate issue:
```
$ python test_subscribe_and_retained_msgs.py
testClientJDH
testClientJDH Connected with result code 0

Subscribe to topic: zm-waterlevel/testJDH
Success
Rx Water Level?

Unsubscribe first
Subscribe to topic: zm-waterlevel/testJDH
Success
Rx Water Level?
zm-waterlevel/testJDH 123

Subscribe to topic: zm-waterlevel/testJDH
Success
Rx Water Level?

Unsubscribe first
Subscribe to topic: zm-waterlevel/testJDH
Success
Rx Water Level?
zm-waterlevel/testJDH 123
```

**Comment by HJianBo** [comment]
@jhor511 Has fixed and released at v2.3.0

**Comment by emqplus** [comment]
@jhor511 I close the issue first. Please install the v2.3.0 and verify it.
